### ISSUE TITLE
Newly added 'fan' devices show up as air conditioners in HomeKit

### ISSUE THREAD

**Issue by einichi** [lprhodes/homebridge-broadlink-rm#616]
Homebridge 1.0.4
homebridge-broadlink-rm 3.6.22

Also tried rolling back to a previous VM snapshot that had Homebridge 0.4.53 and the same homebridge-broadlink-rm 3.6.22, same issue occurs.

Each time I removed the bridge from my HomeKit and when I add the bridge back, it shows up as in the Home app as an aircon device with 'heat', 'cool', 'auto', 'off' options with the temperature slider thing.

Here's my config excerpt for the fan:
```
        {
            "platform": "BroadlinkRM",
            "name": "Broadlink RM",
            "hideScanFrequencyButton": true,
            "hideLearnButton": false,
            "hideWelcomeMessage": true,
            "accessories":  {
                    "name": "Living Fan",
                    "type": "fan",
                    "hideSwingMode": true,
                    "hideRotationDirection": true,
                    "data": {
                        "on": "260026016a370e0d0d0e0c290d0e0e280d0d0d0e0e0d0d280e280c0f0c0f0c0e0e280d280d0e0e0d0c0f0c0e0e0d0e0d0d0e0c0f0c0f0c0e0e0d0d0e0c0e0e280d0e0d0d0e0d0d0e0e0d0c0f0d0d0d290d0e0c0e0d0e0d0e0d290c290e280c290d290d280e280d280e280d0e0d280d0e0d0e0d0d0d0e0e280d0d0e0d0d0e0e0d0c0f0c290e0d0c0f0d0d0e0d0d0e0d0e0d0d0d0e0e280d280e280e0d0c0e0e0d0d0e0d0e0c0f0d280e0d0d0e0d0d0e0d0e280d0d0d0e0d0b0f0f0d0e0c0e0e0d0e0d0d0e0d0d0d0e0e0d0c0f0d0e0d0e0c0e0e0d0c0f0c0e0e0d0e0d0c0f0c0e0e0d0e0d0c0f0c0e0e0d0d0e0e0d0d0e0c0e0d0e0d0e0c0f0d280e0d0d0e0d280d0e0d290d0d0e280d0e0d0d0e0d0e0d0d0e0d0e0c0e0e0d0d0e0c0f0c290e0d0e0d0d280e280c000d050000",
                        "fanSpeed20": "260026016a370e0d0d0e0c290d0e0e280d0d0d0e0e0d0d280e280c0f0c0f0c0e0e280d280d0e0e0d0c0f0c0e0e0d0e0d0d0e0c0f0c0f0c0e0e0d0d0e0c0e0e280d0e0d0d0e0d0d0e0e0d0c0f0d0d0d290d0e0c0e0d0e0d0e0d290c290e280c290d290d280e280d280e280d0e0d280d0e0d0e0d0d0d0e0e280d0d0e0d0d0e0e0d0c0f0c290e0d0c0f0d0d0e0d0d0e0d0e0d0d0d0e0e280d280e280e0d0c0e0e0d0d0e0d0e0c0f0d280e0d0d0e0d0d0e0d0e280d0d0d0e0d0b0f0f0d0e0c0e0e0d0e0d0d0e0d0d0d0e0e0d0c0f0d0e0d0e0c0e0e0d0c0f0c0e0e0d0e0d0c0f0c0e0e0d0e0d0c0f0c0e0e0d0d0e0e0d0d0e0c0e0d0e0d0e0c0f0d280e0d0d0e0d280d0e0d290d0d0e280d0e0d0d0e0d0e0d0d0e0d0e0c0e0e0d0d0e0c0f0c290e0d0e0d0d280e280c000d050000",
                        "fanSpeed40": "260026016a370d0e0d0e0c290d0e0d290c0f0c0e0d0e0d290c290d0e0c0f0c0f0c290d290c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c290d0e0d0e0c0f0c0e0d0e0d0e0c0f0c290e0d0d0e0c0f0c0f0c290d290c290e280c290d290c290f270c290d0e0d290d0e0c0e0d0e0d0e0c290d0e0d0e0d0e0c0f0c0e0d290d0e0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d290c290d290d0e0c0e0d0e0d0e0c0f0c0e0d290d290c0e0d0e0d0e0c2a0c0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c290e0d0d0e0c2a0d0d0d290c0f0c290e0d0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d290d280d0e0d0e0d0e0c290d290c000d050000",
                        "fanSpeed60": "260026016d350c0e0f0c0d290c0e0d290c0f0e0d0e0c0d290e280e0c0d0e0f0c0c290f270f0c0e0d0d0d0f0c0d0e0e0d0e0c0f0c0d0e0f0c0c0f0e0c0d0e0f270c0f0c0e0d0e0f0c0c0f0e0c0d0e0f270c0f0e0c0f0c0d0e0e280e270d290c290f270e270d290c2a0e270f0c0c290d0e0d0e0d0e0c0f0c290f0c0c0f0e0c0d0e0d0e0f270c0e0d0e0d0e0c0f0e0d0c0e0f0c0d0e0c2a0c290d290e0c0d0e0d0e0c0f0e0d0c0e0f0c0d290c290d0e0d0e0c290d0e0d0e0c0f0c0f0c0e0d0e0f0c0c0f0e0c0d0e0d0e0d0e0c0f0c0e0f0c0d0e0c0f0e0c0f0c0d0e0e0d0c0f0c0e0f0c0d0e0d0e0c0f0e0c0d0e0d0e0c0f0c0f0e0c0d0e0d0e0c290d0e0d0e0d290c0e0f270c0f0c290d0e0e0d0c0f0e0c0d0e0f0c0e0d0c290d290c0f0e0c0d0e0d0e0f270c290f000d050000",
                        "fanSpeed80": "260026016a370e0d0c0f0c290d0e0d290c0f0c0e0d0e0d290c290d0e0d0e0c0f0c290f270c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0f0c0d0e0c2a0c0e0d0e0d0e0e0d0c0f0c0e0d0e0d290c0e0d0e0d0e0d0e0c290d290c290d290d290c290d290c290d290c0f0c290d0e0d0e0c0f0c0e0d290d0e0c0f0c0e0d0e0d0e0c2a0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c290d290d290c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c290d0e0c2a0c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c0f0c0e0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d290c0f0c0e0d290c0f0c290d0e0d290c0e0d0e0d0e0d0e0c0f0c0e0d0e0d290c0f0c0e0d0e0d0e0c0f0c290d290d000d050000",
                        "fanSpeed100": "260076006c360c0f0c0f0c290d0e0c290d0e0d0e0d0e0c290d290c0f0c0f0c0e0d290d290d0d0d0e0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c290d0e0d0e0c0f0c0e0d0e0d0e0d0e0c290e0d0d0e0c0f0c290d0e0d0e0c2a0d280e280d0d0d0e0d0e0c2a0d280d0e0c0f0c0e0d290e280c000d050000",
                        "off": "260076006b370c0f0c0e0d290c0f0c290e0d0d0e0c0f0c290d290d0e0c0e0d0e0d290c290e0d0d0e0c0f0c0e0d0e0d0e0d0e0c0f0c0e0d0e0d0e0c0f0c0f0c290d0e0c0f0c0e0d0e0d0e0d0e0c0f0c290e0d0c0f0c0f0c0e0d290d0e0c0f0c0e0d0e0d0e0c0f0c290d0e0d290d280d290c290e280c2a0d000d050000"
                    }
                },
```

I have other actual aircon devices and many other switch devices working fine with this config.

I also have a separate Robot Vacuum device plugin that emulates a fan device in order to control it, and that is still showing up as a fan just fine.

Could this be the result of a change to the HomeKit API?

**Comment by einichi** [comment]
Possible duplicate of #602

**Comment by einichi** [comment]
Can confirm this resolves with a rollback to 3.6.18 as per @stefanodeg's comment in #602 

Closing as duplicate in favour of #602

**Comment by ffaamm** [comment]
[Good catch, it's a bug introduced in `accessories/fan.js` after 3.6.18.](https://github.com/lprhodes/homebridge-broadlink-rm/commit/ccb3fff578ee24c3673b908eded4073773be6026#diff-e5e9ad8be47ce9d631ec056fc292b99b)

[Even `accessories/garageDoorOpener.js` is affected.](https://github.com/lprhodes/homebridge-broadlink-rm/commit/ccb3fff578ee24c3673b908eded4073773be6026#diff-b85751f6b8babb4f681ea5754e7615b7)
